lib/manageiq/appliance_console/message_configuration.rb

```diff
@@ -0,0 +1,134 @@
+require 'active_support/core_ext/module/delegation'
+require 'pathname'
+
+module ManageIQ
+  module ApplianceConsole
+    class MessageConfiguration
+      attr_reader :username, :password,
+                  :miq_config_dir_path, :config_dir_path, :sample_config_dir_path,
+                  :tools_log4_properties_path, :tools_log4_properties_sample_path,
+                  :client_properties_path,
+                  :keystore_dir_path, :truststore_path, :keystore_path,
+                  :messaging_yaml_sample_path, :messaging_yaml_path
+
+      BASE_DIR                          = "/opt/kafka".freeze
+      LOGS_DIR                          = "#{BASE_DIR}/logs".freeze
+      CONFIG_DIR                        = "#{BASE_DIR}/config".freeze
+      SAMPLE_CONFIG_DIR                 = "#{BASE_DIR}/config-sample".freeze
+      MIQ_CONFIG_DIR                    = "/var/www/miq/vmdb/config".freeze
+
+      def initialize(options = {})
+        @username, @password = nil
+
+        @username = options[:username] || "admin"
+        @password = options[:password]
+
+        @miq_config_dir_path               = Pathname.new(MIQ_CONFIG_DIR)
+        @config_dir_path                   = Pathname.new(CONFIG_DIR)
+        @sample_config_dir_path            = Pathname.new(SAMPLE_CONFIG_DIR)
+
+        @tools_log4_properties_path        = config_dir_path.join("tools-log4j.properties")
+        @tools_log4_properties_sample_path = sample_config_dir_path.join("tools-log4j.properties")
+        @client_properties_path            = config_dir_path.join("client.properties")
+
+        @keystore_dir_path                 = config_dir_path.join("keystore")
+        @truststore_path                   = keystore_dir_path.join("truststore.jks")
+        @keystore_path                     = keystore_dir_path.join("keystore.jks")
+
+        @messaging_yaml_sample_path        = miq_config_dir_path.join("messaging.kafka.yml")
+        @messaging_yaml_path               = miq_config_dir_path.join("messaging.yml")
+      end
+
+      def create_tools_log_config
+        say(__method__.to_s.tr("_", " ").titleize)
+
+        FileUtils.cp(tools_log4_properties_sample_path, tools_log4_properties_path) unless file_found?(tools_log4_properties_path)
+      end
+
+      def create_client_properties
+        say(__method__.to_s.tr("_", " ").titleize)
+
+        content = <<~CLIENT_PROPERTIES
+          ssl.truststore.location=#{truststore_path}
+          ssl.truststore.password=#{password}
+
+          sasl.mechanism=PLAIN
+          security.protocol=SASL_SSL
+          sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required \\
+            username=#{username} \\
+            password=#{password} ;
+        CLIENT_PROPERTIES
+
+        File.write(client_properties_path, content) unless file_found?(client_properties_path)
```

Comment:
Should we `return unless ...` earlier to avoid creating the heredoc or should we always write the file in case the values changed?

Reply:
It seems to be a balance of one extra line of code in the source or creating the heredoc.
I'll avoid the heredoc.
Addressed by commit: https://github.com/ManageIQ/manageiq-appliance_console/pull/130/commits/48209af0a4f77eff5d9286b3617d0dea94fad40d